src/NexusMods.DataModel/GameRegistry/Registry.cs

```diff
@@ -130,4 +147,29 @@ public Task StopAsync(CancellationToken cancellationToken)
     {
         return Task.CompletedTask;
     }
+    
+    private void WaitUntilInitialized()
+    {
+        /*
+            Note(sewer)
+            
+            This is an IHostedService.
+            It will be initialized in a background threadpool thread.
+            
+            Sleeping current thread is OK as this can't be called from
+            the thread initializing this.
+            
+            Extra Note:
+            
+            On Windows the Timer Resolution means we can only sleep
+            at minimum 15.6ms increments unless we request a lower one.
+            
+            So expect ~16ms sleeps there.
+        */
+#if DEBUG
+        Debug.Assert(_initThreadId != Environment.CurrentManagedThreadId, "WaitUntilInitialized called from initialization thread");
+#endif
+        while (!_isInitialized)
+            Thread.Sleep(8);
```

Comment:
> Any chance we are sleeping on the Main thread?

Practically speaking, no chance. The service starts in the background as the App's main process is booted up.

By the time the Avalonia window is even displayed, the background initialization should already have finished. The operation being waited for takes around as long as the sleep is set to.

Currently this is only reproducible in CLI commands when a new main process has to be started.

> Should we use an Async Lock?
> Should we rethink how this registry works?

The perf difference is negligible enough that it shouldn't really matter.

Technically speaking though, I could spin a ThreadPool thread (Task.Run) in the constructor and just wait for that to finish. That could delay app startup by a few millis then, not sure if worth it since the APIs here aren't very hot.